docker-compose.yml

```diff
@@ -0,0 +1,39 @@
+version: "3.4"
+
+services:
+  localstack-s3:
+    image: localstack/localstack:latest
+    container_name: localstack-s3
+    environment:
+      - SERVICES=s3:5002
+      - PORT_WEB_UI=8081
+      - DEFAULT_REGION=eu-west-2
+      - DATA_DIR=/tmp/localstack/data
+    ports:
+      - "5002:5002"
+      - "8081:8081"
+
+  db:
+    image: postgres
+    environment:
+      POSTGRES_USER: postgres
+      POSTGRES_PASSWORD: example
+      POSTGRES_DB: airflow-db
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      interval: 30s
+      timeout: 30s
+      retries: 3
+
+  app:
+    build:
+      dockerfile: Dockerfile
+      context: .
+      args:
+        - env=dev
+    volumes:
+      - ./airflow.cfg:/src/airflow/airflow.cfg
+    ports:
+      - 8080:8080
```

Comment:
any `healthcheck` here?

Reply:
do you have a suggestion for checking the health status of this container?

Reply:
`curl localhost:8080/`? Assuming `curl` or `wget` or similar is available, and `/` is an existing path - to be verified.

Reply:
`curl` or `wget` are not installed by default. Would you like me to add them to add a health check? should I add a health check to `localstack` also?

Reply:
The `app` seems designed to wait for the other services, so it's the only one where I see the healthcheck as strictly necessary so that we can deploy, [wait for the container to be healthy](https://github.com/libero/scripts/blob/master/docker/wait-healthy.sh) and run smoke tests. Depending on the application, sometimes checking the port is open with `netcat` is enough; sometimes a real HTTP request is needed which would mean installing the missing `curl`/`wget`.

Reply:
There isn't a `netcat` in the container, so this is the best I have found:
```
bash -c 'echo > /dev/tcp/localhost/8080'
```
It's a bash-ism to check a port is open; assumes if the port has been opened the webserver is ready to answer requests.